python/pyiceberg/catalog/glue.py

```diff
@@ -0,0 +1,246 @@
+
+
+class GlueCatalog(Catalog):
+
+    def __init__(self, name: str, **properties: Properties):
+        super().__init__(name, **properties)
+        self.glue = boto3.client("glue")
+
+    def _default_warehouse_location(self, database_name: str, table_name: str):
+        try:
+            response = self.glue.get_database(Name=database_name)
+        except self.glue.exceptions.EntityNotFoundException:
+            raise NoSuchNamespaceError(f"The database: {database_name} does not exist")
+
+        if PROP_GLUE_DATABASE_LOCATION in response[PROP_GLUE_DATABASE]:
+            return f"{response[PROP_GLUE_DATABASE][PROP_GLUE_DATABASE]}/table_name"
+
+        if PROP_WAREHOUSE in self.properties:
+            return f"{self.properties[PROP_WAREHOUSE]}/{database_name}.db/{table_name}"
+
+        raise ValueError("No default path is set, please specify a location when creating a table")
+
+    def _resolve_table_location(self, location: Optional[str], database_name: str, table_name: str):
+        if not location:
+            return self._default_warehouse_location(database_name, table_name)
+        return location
+
+    # tested on pre-existing database
+    def create_table(
+            self,
+            identifier: Union[str, Identifier],
+            schema: Schema,
+            location: Optional[str] = None,
+            partition_spec: PartitionSpec = UNPARTITIONED_PARTITION_SPEC,
+            sort_order: SortOrder = UNSORTED_SORT_ORDER,
+            properties: Properties = EMPTY_DICT,
+    ) -> Table:
+        """Create a table
+
+        Args:
+            identifier: Table identifier.
+            schema: Table's schema.
+            location: Location for the table. Optional Argument.
+            partition_spec: PartitionSpec for the table.
+            sort_order: SortOrder for the table.
+            properties: Table properties that can be a string based dictionary.
+
+        Returns:
+            Table: the created table instance
+
+        Raises:
+            AlreadyExistsError: If a table with the name already exists
+            ValueError: If the identifier is invalid
+        """
+        database_name, table_name = self.identifier_to_tuple(identifier)
+
+        location = self._resolve_table_location(location, database_name, table_name)
+        metadata_location = f"{location}/metadata/00000-{uuid.uuid4()}.metadata.json"
+        metadata = new_table_metadata(
+            location=location, schema=schema, partition_spec=partition_spec, sort_order=sort_order,
+            properties=properties
+        )
+        io = load_file_io({**self.properties, **properties}, location=location)
+        _write_metadata(metadata, io, metadata_location)
+        try:
+            self.glue.create_table(
+                DatabaseName=database_name,
+                TableInput={
+                    'Name': table_name,
+                    'Description': properties[PROP_TABLE_DESCRIPTION]
```

Comment:
is description required to create the Glue table? If not we should use None instead of ""

Reply:
Ah, I see. Thank you for your suggestion. Since the `Description` field does not accept `None` directly, I made a helper method called `_construct_table_input` to implement the logic here. I think the added method can make the code structure clear when we want to add support for `StorageDescriptor` in the future.